### PR TITLE
[Performance] Ensure ReflectionOperations return `Attribute[]` when possible

### PR DESCRIPTION
**Pull request by Youssef1313**
Motivation:

image

### CODE REVIEW

File: src/Adapter/MSTestAdapter.PlatformServices/Services/ReflectionOperations.cs [hunk 1]
```diff
@@ -60,6 +72,6 @@ public object[] GetCustomAttributes(Assembly assembly, Type type) =>
 #if NETFRAMEWORK
         ReflectionUtility.GetCustomAttributes(assembly, type).ToArray();
 #else
-        assembly.GetCustomAttributes(type).ToArray<object>();
```
Comment (Evangelink) [inline src/Adapter/MSTestAdapter.PlatformServices/Services/ReflectionOperations.cs:63]: Isn't this line changing behavior by setting inherit to true?

**Review by Evangelink — COMMENTED** [review]
Would be good to have a before/after on the allocation screenshot

File: src/Adapter/MSTestAdapter.PlatformServices/Services/ReflectionOperations.cs (hunk 1 above)
Comment (Youssef1313) [reply src/Adapter/MSTestAdapter.PlatformServices/Services/ReflectionOperations.cs:63]: That is the default already

https://github.com/dotnet/runtime/blob/2d81cbb28a8feb1af5b790664b511b264d032877/src/coreclr/System.Private.CoreLib/src/System/Attribute.CoreCLR.cs#L697-L701

Comment (Youssef1313) [reply src/Adapter/MSTestAdapter.PlatformServices/Services/ReflectionOperations.cs:63]: And actually, for assemblies specifically, this parameter has no meaning and is unused.

https://github.com/dotnet/runtime/blob/2d81cbb28a8feb1af5b790664b511b264d032877/src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs#L332-L340

**Review by Evangelink — APPROVED** [review]

**Pull request merged**